### ISSUE TITLE
[FIXED] [REFERENCE] Broken patch on current LineageOS 13 builds (20/10)

### ISSUE THREAD

**Issue by rafifos** [FlymeOS/devices-base_cm#21]
At some point, the patch in com.android.server.wm.WindowManagerService.addWindow broke. Causing it to not boot anymore, the addWindow method is being rejected at boot-time.

Here is the paste of the output on logcat: https://ghostbin.com/paste/w5gt3

TLDR; [0x4BD] monitor-exit on non-object (Undefined)

Here is a tree to test with: https://gitlab.com/h4ndshake/flyme_devices_motorola_ghost
TLDR²; (The method is located at 14478-16533) https://gitlab.com/h4ndshake/flyme_devices_motorola_ghost/blob/72969ae206f5c3b7d79229858ad7d0dd574cb4e1/services.jar.out/smali/com/android/server/wm/WindowManagerService.smali

**Comment by aymende7** [comment]
+1

**Comment by rafifos** [comment]
As a TEMPORARY fix, you can copy the entire method from this repo to your patchrom tree.. it worked here.

**Comment by aymende7** [comment]
i am always getting this error while building 
out/obj/system/framework/services.a7s/smali/com/android/server/policy/PhoneWindowManager.smali[9706,4] There is already a label with that name.
Exception in thread "main" brut.androlib.AndrolibException: Could not smali file: com/android/server/policy/PhoneWindowManager.smali

any ideas?

**Comment by rorre** [comment]
Got this on your method
```monitor-exit v33```
this should be ```monitor-exit v34``` right?

EDIT: explaination, there is no ```monitor-enter v33``` on your method that's why its undefined.

**Comment by rafifos** [comment]
On mine? where?

**Comment by rorre** [comment]
Line 15967

**Comment by rafifos** [comment]
I will revert the patch I've made and test it, just give me about an hour..

I'm an absolute beginner at smali :'v

**Comment by rafifos** [comment]
It fixed the method, thanks.

I'm not closing this issue, if anyone have the same problem, it's easier to find the solution.

TLDR; https://github.com/FlymeOS/devices-base_cm/issues/21#issuecomment-340127567 fixes it.
